llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -4999,6 +5000,26 @@ foreach dim = [1, 2, 3, 4, 5] in {
   }
 }
 
+// Intrinsics for Prefetch and Prefetchu
+foreach level = ["L1", "L2"] in {
+   foreach addr = ["global", "local", ""] in {
+     foreach evict = !if(!eq(addr, "global"), ["evictlast", "evictnormal"], ["evictnormal"]) in {
+      defvar suffix = "" # !if(!eq(addr, ""), "", addr # "_") # level # "_" # evict;             
```

Comment:
Do we need the initial `"" #` ?


String join with, possibly optional empty parts can be a useful primitive, in general. 
We may want to extract it into a helper record. E.g. something like this: https://godbolt.org/z/Mer7nK6j5

```
class Join<string sep, list<string> lst> {
  string ret = !foldl("", lst, a, b, !if(!eq(a, ""), b, !if(!eq(b,""), a, !strconcat(a, sep, b))));
}

def A1 : Join<"_", ["a","b","c","d"]>;
def A2 : Join<":", ["","b","","d"]>;

```
--> 
```
def A1 {        // Join
  string ret = "a_b_c_d";
}
def A2 {        // Join
  string ret = "b:d";
}
```

Reply:
Thanks for this tip. Addressed.